docs/plugin/subsystems.md

```diff
@@ -0,0 +1,288 @@
+Thunder provides "Subsystems" which are abstract categories of functionality (Network, Graphics, Internet) that can be marked as active/inactive by plugins. This mechanism allows plugins to delay activation until certain conditions are met, or ensure that plugins are deactivated in a suitable order.
+
+For example, a web-browser plugin could be prevented from starting until the Internet and Graphics subsystems are ready. The Graphics subsystem could be marked as active by a network management plugin, and the graphics by a window manager/compositor plugin.
+
+The state of each subsystem is tracked by the framework and can be queried or modified by individual plugins.
+
+
+## Supported SubSystems
+
+Thunder supports the following subsystems (enumerated in `Source/plugins/ISubSystem.h`)
+
+| **Subsystem** | **Description**                                              |
+| ------------- | ------------------------------------------------------------ |
+| PLATFORM      | Platform is available                                        |
+| SECURITY      | A security system can validate external requests (JSONRPC/WebRequest) |
+| NETWORK       | Network connectivity has been established.                   |
+| IDENTIFIER    | System identification has been accomplished.                 |
+| GRAPHICS      | Graphics screen EGL is available.                            |
+| INTERNET      | Network connectivity to the outside world has been established. |
+| LOCATION      | Location of the device has been set.                         |
+| TIME          | Time has been synchronized.                                  |
+| PROVISIONING  | Provisioning information is available.                       |
+| DECRYPTION    | Decryption functionality is available.                       |
+| WEBSOURCE     | Content exposed via a local web server is available.         |
+| STREAMING     | Content can be streamed.                                     |
+| BLUETOOTH     | The Bluetooth subsystem is up and running.                   |
+
+All subsystems have a negated equivalent - e.g. `NOT_PLATFORM` and `NOT_SECURITY` that indicates the absence of those subsystems.
+
+When the WPEFramework process starts, Controller will check to see which subsystems are provided by plugins (as defined in the plugin metadata or the Controller configuration). These subsystems are sometimes referred to "external", since their lifetime is managed by a plugin external to the Thunder core. 
+
+If the subsystem is not provided by a plugin, then Controller will mark the subsystem as active at startup. The opposite happens at shutdown, with Controller marking the subsystems as inactive.
+
+### Subsystem Metadata
+
+!!! warning
+	Subsystem metadata is currently only accessible over COM-RPC, there are no corresponding JSON-RPC interfaces or datatypes generated for the subsystem COM-RPC interfaces
```

Comment:
At least the status of all the status of all subsystems is available over JSONRPC. See: https://github.com/rdkcentral/Thunder/blob/master/Source/WPEFramework/doc/ControllerPlugin.md#property.subsystems

Reply:
Interesting ;-) I continued reading and than later on you mention that you can get the JSONRPC status fo the susbsytems via the JSONRPC..